### ISSUE TITLE
Feature request: Disable inline sourcemaps on dev builds

### ISSUE THREAD

**Issue by quigleyj-mavenomics** [jupyterlab/jupyterlab#6804]
Recently I ran into an issue with lab builds where, due to large chunks of data being included in the build, the size of the sourcemaps ballooned to 40mb. This would not normally be more than a extension developer annoyance, but with inline sourcemaps the main JS build size grew to 72mb.

#6700 introduced this behavior with `devtool: inline-source-map` on by default in dev builds, which are currently the default for `jupyter lab build`.

Having a way to switch back to adjacent sourcemaps (either by default or by a flag) would alleviate this. Having it default back to adjacent sourcemaps would be my preference to minimize impact on end-users, though.

As a workaround, other people running into this can edit `$prefix/lib/site-packages/jupyterlab/staging/webpack.config.js`:

```diff
-    devtool: 'inline-source-map',
+    devtool: 'source-map',
```

This will hold until the next time the Jupyterlab python package is updated.

cf. https://gitter.im/jupyterlab/jupyterlab?at=5d2669a8cdb70805c97350a7

**Comment by telamonian** [comment]
On the surface, it looks like the best solution to the problem would be for `jupyter lab build` to default to a production build.

@ian-r-rose @blink1073 For an end user (eg someone who install jlab using `pip install jupyterlab`), when exactly is `jupyter lab build` used? During the initial setup/install? When they install an extension?

In any case, the difference between a dev build and a production build is currently pretty slight. It's just the contents of [`jupyterlab/staging/webpack.prod.config.js`](https://github.com/jupyterlab/jupyterlab/blob/master/jupyterlab/staging/webpack.prod.config.js), right? Is there any reason we shouldn't default to production for `jupyter lab build`?

**Comment by saulshanabrook** [comment]
@telamonian Could we just switch the dev build to also use `source-map` instead of `inline-source-map`?

**Comment by telamonian** [comment]
> Could we just switch the dev build to also use `source-map` instead of `inline-source-map`?

Definitely also an option. Based on the info [here](https://survivejs.com/webpack/building/source-maps/)), the advantages of `inline` are:

- Better performance when debugging
- Makes remote debugging easy

so probably nothing critical. Still, my preference would be that we figure out how to appropriately differentiate `jupyter lab build` from any actual dev build. In addition `devtool: 'source-map'`,  there's likely a bunch of other optimizations that we could be making right now but aren't.

**Comment by jasongrout** [comment]
Is this fixed in #6907?

**Comment by telamonian** [comment]
@jasongrout It should be fixed, but the changes from #6907 haven't been included in a release yet. So I guess once `1.1.0` has been published, then we can say this is fixed.

**Comment by jasongrout** [comment]
Our procedure is that once it is merged into master, the issue is fixed and closed. I commented separately on #6907 to see if we should open another PR to backport to 1.0.x.
